### ISSUE TITLE
Support for Azure SQL Database Geo-Zone-Redundant Backup Storage

### ISSUE THREAD

**Issue by AErmie** [hashicorp/terraform-provider-azurerm#19542]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Description

If you want to deploy an Azure SQL Database using Terraform, not all backup storage redundancy options are supported.

The options that are supported in the Azure Portal, include:
- Locally-Redundant backup storage
- Zone-Redundant backup storage
- Geo-Redundant backup storage
- Geo-Zone-Redundant backup storage

![Azure-Portal-Backup-Storage-Redundancy](https://user-images.githubusercontent.com/10376796/205376230-9f38400f-cef2-469e-82a0-12bd1988fd24.png)

When you define the code block in Terraform for the `azurerm_mssql_database` resource, there is a property for `storage_account_type` (shown in the screenshot below).

![Terraform-Code](https://user-images.githubusercontent.com/10376796/205376619-770aa2e5-d6e0-4e15-adfe-712dffc975b9.png)

However, the only supported types are `geo`, `local`, or `zone`, per the error shown by Terraform. `Geo-Zone` is not a supported account type, although it is an option in the portal.

![Terraform-Plan-Error](https://user-images.githubusercontent.com/10376796/205376663-66d4e84b-4b87-4133-99ba-b1170ae92f61.png)

Even though the [azurerm_storage_account](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account) resource itself has a property for [account_replication_type](https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#account_replication_type), which has valid options of: LRS, GRS, RAGRS, ZRS, GZRS and RAGZRS, you cannot use `GZRS` as a `storage_account_type` in the `azurerm_mssql_database` resource either.

It would be appreciated if the Azure Terraform Provider could be updated to support all properties and configurations for this Azure service.

## Other Details

Terraform version: 0.15.5
AzureRM Provider version: 3.34.0

### New or Affected Resource(s)/Data Source(s)

azurerm_mssql_database

### Potential Terraform Configuration

```hcl
resource "azurerm_mssql_database" "this" {
  storage_account_type    = "Geo-Zone"
}
```

### References

- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/mssql_database
- https://registry.terraform.io/providers/hashicorp/azurerm/latest/docs/resources/storage_account#account_replication_type

**Comment by gkello** [comment]
That missing option is currently blocking me to provision HyperScale database with geo-redundant storage and zone_redundant = true: And it is not possible to adjust backup storage type after database creation for HyperScale tier. 
```
resource azurerm_mssql_database "hyperscale" {
  name                  = "example-hyperscaledb"
  server_id             = azurerm_mssql_server.example.id
  sku_name              = "HS_Gen5_2"

  read_replica_count    = 1
  zone_redundant        = true
}
```
results with

Error: waiting for create/update of Database: (Name "example-hyperscaledb" / Server Name "example-mssqlserver" / Resource Group "example-database-rg"): Code="ProvisioningDisabled" Message="Provisioning of zone redundant Hyperscale database with geo backup redundancy is not supported. Zone redundant Hyperscale databases must use either zone or geo zone backup redundancy."
with azurerm_mssql_database.hyperscale
on hyperscaledb.tf line 24, in resource "azurerm_mssql_database" "hyperscale":

**Comment by aparker-cityworks** [comment]
This affects resource azurerm_mssql_managed_instance in the same way.

**Comment by MichalSino** [comment]
Any changes about this ?

**Comment by MichalSino** [comment]
I found that API 23-02-01-preview supports: Local, Zone, Geo and GeoZone, so may be the provider shloud be updated to this Api version.

**Comment by richsage** [comment]
Looking at the code it looks like it already uses API 2023-02-01-preview, and there is a `GeoZone` storage option available - I haven't tested it but maybe this is already solved?

**Comment by MichalSino** [comment]
I confirmed it. It works now. :)

**Comment by katbyte** [comment]
this seems to have been fixed by #23721 - as such i am going to close it out!
